src/lib/OS/SyscallHandler.cc

```diff
@@ -666,16 +735,65 @@ void SyscallHandler::handleSyscall() {
       stateChange = {ChangeType::REPLACEMENT, {currentInfo_.ret}, {0ull}};
       break;
     }
+    case 233: {  // madvise
+      // madvise is not supported
+      // always return zero to indicate success
+      stateChange = {ChangeType::REPLACEMENT, {currentInfo_.ret}, {0ull}};
+      break;
+    }
     case 235: {  // mbind
       // mbind is not supported due to all binaries being single threaded.
       // Always return zero to indicate success
       stateChange = {ChangeType::REPLACEMENT, {currentInfo_.ret}, {0ull}};
       break;
     }
     case 261: {  // prlimit64
-      // TODO: Functionality temporarily omitted as it is unused within
-      // workloads regions of interest and not required for their simulation
-      stateChange = {ChangeType::REPLACEMENT, {currentInfo_.ret}, {0ull}};
+      pid_t pid = currentInfo_.registerArguments[0].get<pid_t>();
+      int resource = currentInfo_.registerArguments[1].get<int>();
+      uint64_t newLimit = currentInfo_.registerArguments[2].get<uint64_t>();
+      uint64_t oldLimit = currentInfo_.registerArguments[3].get<uint64_t>();
+      int64_t retVal = 0;
+      if (pid != 0) {
```

Comment:
Can the else not be removed in favour of:
```
if (pid != 0) {
    stateChange = {ChangeType::REPLACEMENT, {currentInfo_.ret}, {-EPERM}};
    break;
}
```

Reply:
Is there any particular advantage to doing this?

Reply:
not really, looks a bit clean i guess

Reply:
Will leave as is given current solution is slightly more readable in my opinion